### ISSUE TITLE
XMPP: Post to PubSub

### ISSUE THREAD

**Issue by sjehuda** [moinwiki/moin#1686]
Greetings!

In XMPP, using PubSub(1) (XEP-0060), it is possible to store posts as Atom Syndication(2) entries into XMPP(3).

There are several XMPP extensions that apply it, namely XEP-0227 (4) and XEP-0472 (5) which make node `xmpp:microblog:0` as a common choice to manage news for client accounts (e.g. `blog@moinmo.in`), yet a component account (e.g. `pubsub.moinmo.in`) may use all of its nodes for this purpose.

```
pubsub.moinmo.in/
|
|-- updates/
    |-- Sombre
     `-- Varied
|-- releases/
 `-- events/
```

```
pubsub service/
|
|-- node/
    |-- item
     `-- item
|-- node/
 `-- node/
```

I have made a script that realizes viewing PubSub node items as Atom XML feeds over HTTP.

1. [XEP-0060: Publish-Subscribe](https://xmpp.org/extensions/xep-0060.html)
2. [RFC 4287: The Atom Syndication Format](https://www.rfc-editor.org/rfc/rfc4287)
3. [Atom Over XMPP](https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf)
4. [XEP-0277: Microblogging over XMPP](https://xmpp.org/extensions/xep-0277.html)
5. [XEP-0472: Pubsub Social Feed](https://xmpp.org/extensions/xep-0472.html)
6. [PubSub Over HTTP](https://codeberg.org/poezio/slixmpp/issues/3526)

**Comment by ThomasWaldmann** [comment]
Maybe don't assume that people are familiar with this and describe it from a "how is this useful" perspective.

**Comment by sjehuda** [comment]
XMPP -> PubSub (XML) -> 

There are several observations:

_Note: I am well aware that VPS can coast as low as $3 a month._

1. People who do not want a website, or have content they do not want to publish on an HTTP website, and want to utilize a moin CMS and associate their content with their personal XMPP account, either by making it publicly available or only for their contacts to read.
2. XMPP server administrators who would want to:
(a) Use their XMPP instances as a CMS, and so they can use dedicated PubSub nodes where they publish news that are of the interest to their users.
(b) Provide their XMPP instances as an extra service of which people who hold XMPP account on those servers would have a XMS system of their own.
3. Portability: Sharing data over PubSub (namely in Atom Syndication Format) would provide a standard fashion to convey data to any other platform from it, be it to Movim or to a [feed reader](https://github.com/lwindolf/liferea/discussions/1356) (provided it can ommunicate with XMPP).

Mr. Kevin Cox of [statit](https://gitlab.com/kevincox/statit/-/issues/1) has made the following argument of post processing (after deployment):

> I don't see the advantage of integrating this into statit.
>
> I have a very similar workflow with pinging WebSub hubs about the feeds after a deploy. However I just do this with a curl command after the deploy step.

For which I have suggested to use PubSub nodes as both a mean of publishing and a mean of storing source files, which makes it also as a backup and synchronization solution:

```
Publishing node <- Public or restricted to certain JIDs
|- Item (ID: 0001)
  |- Atom entry (HTML inside)
|- Item (ID: 0002)
|- Item (ID: 0003)
|- 
...

Management node <- Private only to the account owner.
```
|- Item (ID: 0001)
  |- Source file
|- Item (ID: 0002)
|- Item (ID: 0003)
|- ...
```

Currently, only two platforms are offering a user-friendly view of PubSub are Movim and Libervia and both are web based.
Yet, the more CMS software to implement PubSub, the more desktop clients would add support for PubSub.

The basic [solution](https://codeberg.org/poezio/slixmpp/issues/3526) I offer is not the best, yet is sensible, i na form of providing a web interface to fetch content from pubsub node and display it as HTML or as XML (converted into HTML using an XSLT stylesheet) which is readable with a feed reader.

**Comment by RogerHaase** [comment]
I think @sjehuda is an AI experiment. Are there bots filling GitHub with fake projects and creating fake issues on existing projects?

**Comment by sjehuda** [comment]
@RogerHaase, good day (it is evening here)

My name is Schimon Jehudah and I am an attorney at law.

### Concerning to myself

While understanding the issues of Javascript in recent years, especially after the HTML5 propaganda and the joint removal of syndication (i.e. Atom/RDF/RSS) support from corporate web browser, I am working furiously to think of ideas of recoverying from this issue.

### Concerning to my projects

In the last couple of years, I have been working on a news syndication bot for XMPP.

https://gitgud.io/sjehuda/slixfeed

I have also made a bookmarks manager project 

https://gitgud.io/sjehuda/slixmark

and many others like a [pure Javascript feed parser](https://greasyfork.org/en/scripts/465932-newspaper-native-web-feed-reader) after I have found out that the red lizard has [sabotaged](https://openuserjs.org/garage/Why_my_script_doesnt_work_with_Firefox#comment-1810ec600fe) its proposed library of XSLTProcessor.

I am also using non-corporate web browsers and so I look up for solutions for popular extension like [ClearURLs](https://openuserjs.org/scripts/sjehuda/Clean_URL_Improved) and [Privacy Redirect](https://openuserjs.org/scripts/sjehuda/Proxy_Redirect).

I am now focusing on CMS software to support PubSub and I have even made a [Python script](https://codeberg.org/poezio/slixmpp/issues/3526) (a first of its kind that realizes the use of slixfeed library being used as an instance which is not the main "actor" (so called) of a Python program.

Due to the urgency of this matter, I have ceased to receive any legal work this year 2024 in order to be more focused on Syndication, XMPP and other important technologies.

The state of which I have set myself for 2024 where I live and by the legal system to which I am subjected to, cost me over $500 per month (namely "property tax" and "social security").

### Questions

Please kindly explain to me these:
- If I was fake, why would make my activity here public?
- What made you to determine that I create fake issues?
- What made you to determine that I create fake projects?

Good night,
Schimon

**Comment by sjehuda** [comment]
I have updated my last post.

**Comment by RogerHaase** [comment]
Reported sjehuda's comments as spam, blocked user from project.

**Comment by Alumniminium** [comment]
> Reported sjehuda's comments as spam, blocked user from project.

...why?

**Comment by ThomasWaldmann** [comment]
I don't think @sjehuda is an AI, he contacted me via email.

But I explained to him that:
- project focus for MoinMoin is getting moin2 into a releasable state with a basic feature set
- likely XMPP features are not part of such a basic feature set
- that his posts are hard to read and understand for people unfamiliar with XMPP and should have been written from a more practically motivated POV, like "what problem are we solving, why and how would anyone use such functionality".

So I would suggest to unblock him, so he can edit and improve this or maybe better create one new issue.

I currently don't think this could be in scope for moin2 development until moin 2.1 though.

**Comment by Neustradamus** [comment]
I confirm that @sjehuda is not a bot or a spammer. Please unmark comments.

The feature request is really important!

You can look for example here:
- https://github.com/xenocrat/chyrp-lite/issues/250
- https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/XMPP.20and.20PubSub

For example, you can contact on Psi+ XMPP MUC Room: xmpp:psi-dev@conference.jabber.ru?join

To use XMPP, there is a client list (Desktop, Mobile or Web) here:
- https://xmpp.org/software/

**Comment by RogerHaase** [comment]
I have unblocked @sjehuda and unhide his posts. I apologize for mistaking him as a bot.

**Comment by sjehuda** [comment]
We are all friends here and we are all sharing similar causes!

Please pardon me for the belated respond.

---

Below is a part of a correspondence which I have had with Thomas:

I hope this clearly expains the matter.

> I know a little, but only very little and even I could not extract much from what you wrote, especially not "what problem are we solving here?", practical use cases, why and how somebody would want to use something like what you proposed.

* Managing MoniMoin via XMPP alone (optional).
* Easier access to data.
* Portability of data.
* Uniformity of data (disputable).

Using XMPP as a hosting medium with XML files that MoinMoin already generates.

While HTML pages of MoinMoin are namely dedicated to HTTP browsers, XML can be dedicated to XMPP just as it is dedicated to HTTP feed readers.

This is useful, in general, and in particular to communities revolving around XMPP, such as Gajim, Psi, Speed Dreams and many more who maintain a wiki and want to incorporate Wiki-CMS like MoinMoin with XMPP.

That would make MoinMoin a CMS that can be also viewed and managed from XMPP alone.

---

> * Uniformity of data (disputable).
By "uniformity" I meant to HTML.
Yet, it would be posible to upload both files to two different nodes: One node for source files (to be used for editing) and one for HTML pages (to be used for viewing).

**Comment by sjehuda** [comment]
Gentlemen, greetings!

I have made a dynamic journal generator ([Rivista](https://git.xmpp-it.net/sch/Rivista)) which really was intended to be a simple and efficient Atom feed generator for XMPP PubSub Nodes, and, incidentally, it has turned into a dynamic journal generator.

The screenshots below display contents that are not store on the server used to operate Rivista (you can also start the script and use the address `127.0.0.1:8000`), the contents are stored on various of PubSub Nodes of XMPP servers.

Currently, it is only possible to view content.

I might add a login mechanism to comment on posts and add contents.

![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:01:10 833 webp](https://github.com/user-attachments/assets/a1d36aa8-0f90-45af-a1e3-88862f9327bc)


<details>

![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:01:16 933 webp](https://github.com/user-attachments/assets/5261be2a-0d39-456a-b907-77c03ef537e7)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:01:16 100 webp](https://github.com/user-attachments/assets/0d4748a4-a517-4ad1-ad96-8b9be59ec63c)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:01:15 767 webp](https://github.com/user-attachments/assets/cd920ee7-d43a-4978-a88a-d3615587389c)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:01:06 100 webp](https://github.com/user-attachments/assets/fb993361-b382-4ccd-a93d-e7bde40aebb4)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:00:52 333 webp](https://github.com/user-attachments/assets/f008bb37-2006-48b3-9624-b15b2185b24f)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:00:43 633 webp](https://github.com/user-attachments/assets/e3226602-4d64-404a-895a-11b06c8ff078)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:00:35 867 webp](https://github.com/user-attachments/assets/a3d0ba7f-e81d-484f-8133-dd98dd9780da)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:00:29 300 webp](https://github.com/user-attachments/assets/26b14139-7077-491d-a226-d119e17379b1)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:00:26 367 webp](https://github.com/user-attachments/assets/ea802a56-09e2-4cde-8e97-6c5d46bb9e5d)
![mpv-rivista-plane-jabber-at-wpn-2024-07-14_09 37 04 mkv-00:00:00 000 webp](https://github.com/user-attachments/assets/98f25b09-19ca-4129-8d0a-0ea9e7fd210d)

</details>

**Comment by sjehuda** [comment]
I am currently working on a Python script for Poezio, which would allow me to port into Nikola.

gemini://woodpeckersnest.space/~schapps/projects/poezio-journal.gmi

Nicolas has been working on a Python script for another publishing platform, which can also be ported to Nikola.

https://codeberg.org/slidge/blog/src/branch/main/pubsub.py

**Comment by sjehuda** [comment]
I am currently working on a Python script for Poezio, which would allow me to port into MoinMoin.

gemini://woodpeckersnest.space/~schapps/projects/poezio-journal.gmi

Nicolas has been working on a Python script for another publishing platform, which can also be ported to MoinMoin.

https://codeberg.org/slidge/blog/src/branch/main/pubsub.py

￼

**Comment by sjehuda** [comment]
Mr. Nicolas Cedilnik (nicoco) has created a plugin for Pelican.
* https://github.com/getpelican/pelican/issues/3300
* https://codeberg.org/nicoco/pelican-pubsub
* https://portal.mozz.us/gemini/woodpeckersnest.space/~schapps/journal/2025-03-14-good-news-friday-pelican-joins-to-xmpp.gmi

Code: Python
License: AGPL-3.0
